src/client/components/LoginForm/LoginForm.tsx

```diff
@@ -1,56 +1,120 @@
-import React from 'react';
-import type { FunctionComponent } from 'react';
-
-const LoginForm: FunctionComponent = () => (
-  <form>
-    <div className="form-outline mb-4">
-      <input type="email" id="emailBox" className="form-control" />
-      <label className="form-label" htmlFor="emailBox">
-        Email address
-      </label>
-    </div>
-
-    <div className="form-outline mb-4">
-      <input type="password" id="passwordBox" className="form-control" />
-      <label className="form-label" htmlFor="passwordBox">
-        Password
-      </label>
-    </div>
-
-    <div className="row mb-4">
-      <div className="col">
+import React, { useState, useCallback } from 'react';
+import { useNavigate } from 'react-router-dom';
+import type { FunctionComponent, ChangeEvent, SyntheticEvent } from 'react';
+import { isObjectRecord } from '../../common/utilities/types';
+
+const LoginForm: FunctionComponent = () => {
+  const navigate = useNavigate();
+  // check if user logged in already
+  const user = localStorage.getItem('user');
+  if (user !== null) {
+    navigate('/');
+  }
+
+  // const [error, setError] = useState<string | undefined>(undefined);
+  const [username, setUsername] = useState<string>('');
+  const [password, setPassword] = useState<string>('');
+  const handleUsernameChange = useCallback(
+    (e: ChangeEvent<HTMLInputElement>) => {
+      setUsername(e.target.value);
+    },
+    [setUsername],
+  );
+
+  const handlePasswordChange = useCallback(
+    (e: ChangeEvent<HTMLInputElement>) => {
+      setPassword(e.target.value);
+    },
+    [setPassword],
+  );
+
+  const handleSubmit = useCallback(async(e: SyntheticEvent) => {
+    try {
+      e.preventDefault();
+      const response = await fetch('http://localhost:3000/api/login', {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify({
+          username,
+          password,
+        }),
+      });
+
+      const result: unknown = await response.json();
+      console.log(result);
+
+      if (!isObjectRecord(result)) {
+        throw new Error('Unexpected body type: LoginForm.tsx');
+      }
+      if (typeof result.success !== 'boolean') {
+        throw new Error('success variable not type boolean: LoginForm.tsx');
+      }
+      if (typeof result.isloggedin !== 'boolean') {
+        throw new Error('isloggedin variable not type boolean: LoginForm.tsx');
+      }
+      // store in localStorage
+      if (result.success) {
```

Comment:
May want to handle the `else` for this. What do we want to do if `result.success === false`?